targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -556,82 +454,89 @@ static void register_next_tick()
     __enable_irq();
```

Comment:
@sg- No it shouldn't be a critical section because this operation is time critical. On NRF targets critical section rely on softdevice own critical sections which disable all interrupts except the radio related interrupts. If a radio interrupt is executed for a duration of more than 932 us during this transaction then the value inserted in the capture compare register is incorrect. It would have be possible to invalidate the RTC value if incorrect after it has been set and repeat the transaction from the beginning but it reduce the tick accuracy and introduce a noticeable time drift (in the long run) detrimental for all OS primitives based on time. 

Last time I've made a benchmark of this code, it was taking 132ns to execute, small enough to avoid radio perturbations.